### PR TITLE
[SearchPage] Implement search page

### PR DESCRIPTION
**Pull request by asimqt**
Things were done:
1. Focus the search box when opening
2. Close search form when pressing `esc` key.
3. Implement `SearchPage` component and rename it from just `search` since we already have another `search` for searchForm.
4. Implement cache for fastly.

### CODE REVIEW

File: app/isomorphic/components/basic/search.jsx [hunk 1]
```diff
@@ -52,7 +63,12 @@ class Search extends React.Component {
         <form onSubmit={this.handleSubmit} className="qt-search__form component-wrapper" ref={(searchForm) => this.searchForm = searchForm}>
           <label className="qt-search__form-label">
```
Comment (rvgpl) [inline app/isomorphic/components/basic/search.jsx]: for attribute

File: app/isomorphic/components/basic/search.jsx [hunk 2]
```diff
@@ -52,7 +63,12 @@ class Search extends React.Component {
...
             <span>Search query: </span>
-            <input type="text" value={this.state.searchQuery} onChange={this.handleChange} className="qt-search__form-input" placeholder="What are you looking for?"/>
+            <input type="text" value={this.state.searchQuery}
```
Comment (rvgpl) [inline app/isomorphic/components/basic/search.jsx]: missing id

File: app/server/data-loaders/search-page-data.js [hunk 3]
```diff
@@ -1,10 +1,38 @@
...
-    .then(result => ({
-      stories: result.stories.map(story => story.asJson()),
-      total: result.total,
-      query: query
-    }));
+    .then(result => {
+      const menu = config.layout.menu;
+      const structuredMenu = getNavigationMenuArray(menu);
+      return {
+        stories: getProcessedStories(result.stories, config.layout.menu, config.sections),
```
Comment (rvgpl) [inline app/server/data-loaders/search-page-data.js]: If you are not using menu variable created, I don't see a reason why you need it. -_-

File: app/server/data-loaders/search-page-data.js [hunk 4]
```diff
@@ -1,10 +1,38 @@
...
+exports.loadSearchPageData = function loadSearchPageData(client, query, config) {
   return Story.getSearch(client, {'q': query, 'limit': '20'})
-    .then(result => ({
-      stories: result.stories.map(story => story.asJson()),
-      total: result.total,
-      query: query
-    }));
+    .then(result => {
+      const menu = config.layout.menu;
+      const structuredMenu = getNavigationMenuArray(menu);
```
Comment (rvgpl) [inline app/server/data-loaders/search-page-data.js]: Can we name this display menu? what is structured menu?

File: app/server/data-loaders/search-page-data.js [hunk 5]
```diff
@@ -1,10 +1,38 @@
...
+        query: query
+      }
+    });
+}
+
+function getProcessedStories(stories, menu, sections) {
+  const processedStories = [];
+  // TODO: Optimize
+  stories.forEach(story => {
+    story = story.story;
```
Comment (rvgpl) [inline app/server/data-loaders/search-page-data.js]: Why do you require to do this? `stories` here are items of a collection?

File: app/server/data-loaders/search-page-data.js (hunk 4 above)
Comment (asimqt) [reply app/server/data-loaders/search-page-data.js]: Sure. DisplayName would be good. Structured meant I structured the object to show it on the front end. :)

File: app/server/data-loaders/search-page-data.js (hunk 3 above)
Comment (asimqt) [reply app/server/data-loaders/search-page-data.js]: I am passing the menu variable to `getNavigationMenuArray` function. Will use it here too.

File: app/server/data-loaders/search-page-data.js (hunk 5 above)
Comment (asimqt) [reply app/server/data-loaders/search-page-data.js]: Yes. `stories` are an array with each element containing another `story` object.

Comment (asimqt) [reply app/server/data-loaders/search-page-data.js]: It's not a collection though. There is no `type` and `id` attributes.

File: app/server/data-loaders/search-page-data.js (hunk 4 above)
Comment (asimqt) [reply app/server/data-loaders/search-page-data.js]: I am using `navigationMenu`.

File: app/server/data-loaders/search-page-data.js [hunk 6]
```diff
@@ -1,10 +1,38 @@
...
+    const parentSectionId = story.sections[0]['parent-id'];
+    const parentSection = _.find(sections, function(section) { return section.id === parentSectionId; });
+    if(parentSection) {
+      story['generated-slug'] = parentSection.slug + '/' + story.slug;
+      story['parent-collection'] = parentSection;
```
Comment (ajaymenon0) [inline app/server/data-loaders/search-page-data.js]: Need to have different name instead of `parent-collection`

Comment (asimqt) [reply app/server/data-loaders/search-page-data.js]: Will change this to `parent-section` in all occurences in other loaders in new PR.

**Pull request merged**